src/renderer/store/PSTStore.tsx

```diff
@@ -1,82 +1,65 @@
 import type {
     PstContent,
-    PstEmail,
     PstExtractTables,
-    PstFolder,
 } from "@common/modules/pst-extractor/type";
+import type { ComputedDatum } from "@nivo/circle-packing";
 import { atom, useAtom } from "jotai/index";
 import type { SetStateAction } from "react";
-import { useCallback } from "react";
 
-import type { PstComputed } from "../utils/pst-extractor";
-import { computedRoot } from "../utils/pst-extractor";
+import type { MailViewerObject, ViewerObject } from "../utils/pst-extractor";
+import { isMailViewerObject } from "../utils/pst-extractor";
+
+export type MainInfos = ComputedDatum<ViewerObject<string>>;
+
+export const isMailMainInfos = (
+    mainInfos: MainInfos
+): mainInfos is ComputedDatum<MailViewerObject<string>> =>
+    isMailViewerObject(mainInfos.data);
 
 export interface UsePstStore {
     sentFolder: string;
     setSentFolder: (update: SetStateAction<string>) => void;
+    breadcrumb: string;
+    setBreadcrumb: (update: SetStateAction<string>) => void;
     deletedFolder: string;
     setDeletedFolder: (update: SetStateAction<string>) => void;
-    rootPath: string;
-    setRootPath: (update: SetStateAction<string>) => void;
     pstFile: PstContent | undefined;
     setPstFile: (update: SetStateAction<PstContent | undefined>) => void;
-    computedPst: PstComputed | undefined;
-    setComputedPst: (update: SetStateAction<PstComputed | undefined>) => void;
-    updateComputedPst: (pst: PstFolder, nodeId: string) => void;
     extractTables: PstExtractTables | undefined;
     setExtractTables: (
         update: SetStateAction<PstExtractTables | undefined>
     ) => void;
-    mainInfos: PstEmail | undefined;
-    setMainInfos: (update: SetStateAction<PstEmail | undefined>) => void;
-    depth: number | string;
-    setDepth: (update: SetStateAction<number>) => void;
+    mainInfos: MainInfos | undefined;
+    setMainInfos: (update?: SetStateAction<MainInfos | undefined>) => void;
```

Comment:
It feels weird to have `update` possibly undefined, or other `set` methods

Reply:
jotai acts weird on useState... need to investigate on this but `PstStore` will be completely refactor asap!